src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4291,7 +4292,16 @@ void StorageReplicatedMergeTree::startupImpl()
         /// It means that failed "startup" must not create any background tasks that we will have to wait.
         try
         {
-            shutdown();
+            /// it's important to avoid full shutdown here, because it even tries to shutdown attach thread which was
+            /// designed exactly for this: try to start table if no zookeeper connection available.
+            if (from_attach_thread)
+            {
+                restarting_thread.shutdown(/* part_of_full_shutdown */false);
```

Comment:
What about other stuff that `StorageReplicatedMergeTree::shutdown()` do? Like resetting `session_expired_callback_handler` or deactivating `mutations_finalizing_task`?

Reply:
@alesapin, you probably missed a notification